src/OrleansProviders/Streams/Common/PooledCache/ICacheDataAdapter.cs

```diff
@@ -59,6 +60,24 @@ public interface ICacheDataAdapter<in TQueueMessage, TCachedMessage>
         /// Assignable purge action.  This is called when a purge request is triggered.
         /// </summary>
         Action<IDisposable> PurgeAction { set; }
+
+        /// <summary>
+        /// Handler for when a cachedMessage got purged
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        void SignalMessagePurged(ref TCachedMessage cachedMessage);
+
+        /// <summary>
+        /// Determine action to make on a purgeRequest
+        /// </summary>
+        /// <param name="purgeRequest"></param>
+        void ConsultPurgeRequest(IDisposable purgeRequest);
+
+        /// <summary>
+        /// CachePressureMonitor used on its cache
+        /// </summary>
+        ICachePressureMonitor CachePressureMonitor { set; }
```

Comment:
just to be clear, I'm not suggesting a separate class, just a single method to initialize the 2 things that seems to be initialized in tandem. I didn't originally like a set-only method, but less if there's multiple setters that need to be called just to have the object fully initialized